### DISCUSSION TITLE
Visible Closure Table Column - Parameter Not Filled

### DISCUSSION THREAD

**Discussion by untillnesss** [filamentphp/filament#12239] [Help] [answered]
### Package

Table builder

### Package Version

v3.x.x

### How can we help you?

![image](https://github.com/filamentphp/filament/assets/33585758/5464aca4-1846-4ee4-abd8-9604033f508c)

I want to create a condition where columns can appear or hide according to the data row they pass.

Usually we just enter a callback function with the `$record` parameter to get 1 data in that row, but when I try the `visible()` function, it returns `NULL` or not filled in.

![image](https://github.com/filamentphp/filament/assets/33585758/3584db86-a39e-4234-a62a-d6584f9e0909)

Is there anything I missed?

**Comment by dherrlein** [comment]
You could use html class:

`->extraAttributes(fn($record): array => $record->type === 'spec' ? ['class' => 'hidden'] : []),`

**Comment by wiwa1978** [comment]
Same issue for me on visible()

**Comment by micraux** [comment]
Hi,

Try to tell which model relates to `$record`:
```php
->visible(function(YourModel $record, $state) {
    dd($record, $state)
    ...
```

**Comment by hlspablo** [comment]
any hope on this? its painful how the basic stuff does not work

**Reply by leandrocfe** [reply, marked as answer]
You can't determine the visibility of a column based on its rows. Instead, use the disabled()

**Reply by hlspablo** [reply]
> You can't determine the visibility of a column based on its rows. Instead, use the disabled()

Thank you Leandro, you right, skill issue, and it makes me feel dumb rsrs

**Reply by leandrocfe** [reply]
> > You can't determine the visibility of a column based on its rows. Instead, use the disabled()
> 
> Thank you Leandro, you right, skill issue, and it makes me feel dumb rsrs

No problem ✌️

**Reply by prykris** [reply]
Perhaps I'm a bit more special. But why?

**Reply by tribey2** [reply]
@prykris Because the column applies to the whole table, not just the row in question

**Reply by prykris** [reply]
> @prykris Because the column applies to the whole table, not just the row in question

That makes so much sense. Thanks! :D
